util/ranger/detacher.go

```diff
@@ -345,6 +355,80 @@ func (d *rangeDetacher) detachCNFCondAndBuildRangeForIndex(conditions []expressi
 	return res, nil
 }
 
+// excludeToIncludeForIntPoint converts `(i` to `[i+1` and `i)` to `i-1]` if `i` is integer.
+// For example, if p is `(3`, i.e., point { value: int(3), excl: true, start: true }, it is equal to `[4`, i.e., point { value: int(4), excl: false, start: true }.
+// Similarly, if p is `8)`, i.e., point { value: int(8), excl: true, start: false}, it is equal to `7]`, i.e., point { value: int(7), excl: false, start: false }.
+func excludeToIncludeForIntPoint(p point) point {
+	if !p.excl {
+		return p
+	}
+	if p.value.Kind() == types.KindInt64 {
+		val := p.value.GetInt64()
+		if p.start {
+			if val != math.MaxInt64 {
```

Comment:
If `val == math.MaxInt64`, we can return an information to the caller that this filer is constant false? Then we can handle this information in `ExtractEqAndInCondition` to return `true` in the last return value.